### PR TITLE
Add a re-reading of the configuration when an instance is restarted

### PR DESCRIPTION
**Pull request by better0fdead**
Moved configure to package config.
Added re-reading of the configuration at watchdog level. Currently it changes only watchdog.restartable if u have any other suggestions I am willing to hear them.

### CODE REVIEW

File: cli/configure/config.go [hunk 1]
```diff
@@ -1,60 +1,20 @@
-package modules
+package configure
```
Comment (0x501D) [inline cli/configure/config.go:1]: Rename config.go -> configure.go

File: cli/config/cfg.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package config
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/config/cfg.go:1]: Why config structure is located in separate directiry('config') instead of ('configure') ?

File: cli/running/lua/launcher.lua [hunk 3]
```diff
@@ -61,4 +61,4 @@ local function start_instance()
     return 0
 end
 
-start_instance()
+start_instance()
```
Comment (0x501D) [inline cli/running/lua/launcher.lua:64]: Unnecessary changes

File: cli/running/watchdog.go [hunk 4]
```diff
@@ -27,14 +28,15 @@ type Watchdog struct {
 	restartTimeout time.Duration
 	// done channel used to inform the signal handle goroutine
 	// about termination of the Instance.
```
Comment (0x501D) [inline cli/running/watchdog.go:30]: Comment does not match next variable

File: cli/running/watchdog.go [hunk 5]
```diff
@@ -62,7 +64,11 @@ func (wd *Watchdog) Start() {
 		wd.done <- true
 		// Wait for the signal processing goroutine to complete.
 		wd.doneBarrier.Wait()
-
+		// Parse cfg again to check if values changed.
```
Comment (0x501D) [inline cli/running/watchdog.go:67]: to check that the values have changed

**Review by 0x501D — CHANGES_REQUESTED** [review]
Please resolve conflicts.

File: cli/config/cfg.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/config/cfg.go:1]: It is located in separate directory because modules.go needs it, and configure needs modules, so if config will be part of configure, configure and modules will be dependencies of each other.

File: cli/running/watchdog.go (hunk 5 above)
Comment (better0fdead) [reply cli/running/watchdog.go]: Fixed

File: cli/running/watchdog.go (hunk 4 above)
Comment (better0fdead) [reply cli/running/watchdog.go:33]: Fixed

File: cli/running/lua/launcher.lua (hunk 3 above)
Comment (better0fdead) [reply cli/running/lua/launcher.lua]: Removed

File: cli/configure/config.go (hunk 1 above)
Comment (better0fdead) [reply cli/configure/config.go]: Fixed

File: cli/config/cfg.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/config/cfg.go]: accepted

File: go.mod [hunk 6]
```diff
@@ -39,4 +39,5 @@ require (
 	golang.org/x/sys v0.0.0-20211205182925-97ca703d548d // indirect
 	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+ 	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
```
Comment (0x501D) [inline go.mod:42]: Is this addition belongs to this feature?

Comment (better0fdead) [reply go.mod]: No, fixed

**Review by 0x501D — APPROVED** [review]
LGTM

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi! Thank you for the patchset.
How do you think, it would be better to move `ModuleInfo` to the `configure` package? Or that won't solve the problem.
Mark `tt: restruct project so parsing of cfg is now in configure package` as `Part of #31`

**Review by LeonidVas — COMMENTED** [review]
`watchdog: add re-reading of cfg after instance is killed`
`After instance shutdown added update of wachtdog settings` - Set a dot at the end of the sentence.
Perhaps `When an instance restarts after shutting down` would be better.

**Comment by LeonidVas** [comment]
In fact, we now have options that will be re-read from the config in case the instance is restarted by watchdog. I think we should describe this in the README.

File: cli/running/watchdog.go [hunk 7]
```diff
@@ -62,7 +66,11 @@ func (wd *Watchdog) Start() {
 		wd.done <- true
 		// Wait for the signal processing goroutine to complete.
 		wd.doneBarrier.Wait()
-
+		// Parse config again to check if the values have changed.
+		if wd.configPath != "" {
+			newCliopts, _ := configure.GetCliOpts(wd.configPath)
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/running/watchdog.go]: Fixed

File: cli/config/config.go [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// CliOpts stores information about Tarantool CLI configuration.
+// Filled in when parsing the tarantool.yaml configuration file.
+//
+// tarantool.yaml file format:
+// tt:
+//   modules:
+//     directory: path/to
+//...app:
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/config/config.go]: Fixed

**Comment by better0fdead** [comment]
> Hi! Thank you for the patchset. How do you think, it would be better to move `ModuleInfo` to the `configure` package? Or that won't solve the problem. Mark `tt: restruct project so parsing of cfg is now in configure package` as `Part of #31`

No, that wont solve the problem since we need to call Runcmd from package modules. As for me that structure of project is the best we could do.

**Review by LeonidVas — CHANGES_REQUESTED** [review]
I had a discussion with my colleagues about this task and it looks like rereading the configuration would be logical to implement by starting to use `tt run` instead of reusing `Instance`. I propose to close this ticket by taking the refactoring commits and implement the rereading within the `tt run` [task](https://github.com/tarantool/tt/issues/29).
